tests/test_edge_cases.py

```diff
@@ -2135,6 +2157,7 @@ def __new__(cls, data: int):
             return self
 
     m = MyModel(my_int=IntSubclass(123))
+    # TODO: Is this still the behavior we want in v2? (Currently m.my_int.__class__ is int)
     assert m.my_int.__class__ == IntSubclass
```

Comment:
Question 5 ☝️

Reply:
yes, because in `pydantic-core` we cast the value to a rust `i64`, so the sub-type information is lost.

If you want to keep subtype information you should use an `is-instance` validator - I guess we need a python type structured as an annotation (like `ImportString`) to allow use of the `is-instance` validator, so the syntax here would be something like

```py
class MyModel(BaseModel):
    my_int: IsInstance[int]
    # which is just shorthand for
    my_int: Annotated[int, CheckIsInstance]
```

(`CheckIsInstance` name needs thought)

Reply:
Makes sense to me